README.md
DKImageBrowser
==============

iOS controller to display images (URL or `UIImage`) alongside a strip of thumbnails

[![Build Status](https://travis-ci.org/dkhamsing/DKImageBrowser.svg?branch=master)](https://travis-ci.org/dkhamsing/DKImageBrowser)
[![Version](https://img.shields.io/cocoapods/v/DKImageBrowser.svg?style=flat)](http://cocoadocs.org/docsets/DKImageBrowser)
[![License](https://img.shields.io/cocoapods/l/DKImageBrowser.svg?style=flat)](http://cocoadocs.org/docsets/DKImageBrowser)
[![Platform](https://img.shields.io/cocoapods/p/DKImageBrowser.svg?style=flat)](http://cocoadocs.org/docsets/DKImageBrowser)

![](Assets/demo.gif)

# Installation
## Manual
- Drag `DKImageBrowser/DKImageBrowser` to your project.
- `#import "DKImageBrowser.h"` (or `#import "DKModalImageBrowser.h"`)

## [CocoaPods](https://cocoapods.org/)
``` ruby
platform :ios, '7.0'
pod 'DKImageBrowser'
```

# Usage
Set up the data source array and pass it to an instance of `DKImageBrowser`.

```  objc    
    DKImageBrowser *imageBrowser = [[DKImageStripController alloc] init];
    imageBrowser.DKImageDataSource =  @[ @"http://placekitten.com/230/400",
                                  	     @"http://placekitten.com/240/400"], ];
    [self.navigationController pushViewController:imagesController animated:YES];

```

The data source `DKImageDataSource` can also be a list of `UIImage` objects.

``` objc
    imageBrowser.DKImageDataSource = @[ [UIImage imageNamed:@"400-1.jpg"],
                                        [UIImage imageNamed:@"400-2.jpg"], ];
```
                                  
To present the controller modally, use `DKModalImageBrowser`.

``` objc
    DKModalImageBrowser *modalImageBrowser = [[DKModalImageBrowser alloc] init];    
    // note: use modalImageBrowser.imageBrowser to set data source, customize
    modalImageBrowser.imageBrowser.DKImageDataSource = @[ [UIImage imageNamed:@"400-1.jpg"],
                                                          [UIImage imageNamed:@"400-2.jpg"], ];
    [self presentViewController:modalImageBrowser animated:YES completion:nil];

```

## Swift

``` swift
 	func actionDefault() {
        let imageBrowser = DKImageBrowser()
        imageBrowser.title = "UIImage Data Source"
        
        let imageDataSource: [UIImage] = [
            UIImage(named: "400-1.jpg")!,
            UIImage(named: "400-2.jpg")!,
        ]
        imageBrowser.DKImageDataSource = imageDataSource
        
        self.navigationController?.pushViewController(imageBrowser, animated: true)
    }
    
    func actionCustom() {
        var kittenDataSource: [NSString] = []
        let imageHeight = 416
        let imageWidth = Int(self.view.bounds.size.width - 20)
        
        for index in 1...10 {
            let placeKitten = "http://placekitten.com/\(imageWidth + (index * 2  + 10))/\(imageHeight)" as NSString
            kittenDataSource.append(placeKitten)
        }
        
        let modalImageBrowser = DKModalImageBrowser()
        modalImageBrowser.title = "URL Data Source"
        
        modalImageBrowser.imageBrowser.DKImageDataSource = kittenDataSource
        modalImageBrowser.imageBrowser.DKImagePadding = 5
        modalImageBrowser.imageBrowser.DKThumbnailStripHeight = 110
        modalImageBrowser.imageBrowser.DKBackgroundColor = UIColor.blackColor()
        modalImageBrowser.imageBrowser.DKThumbnailStripPosition = 0
        modalImageBrowser.imageBrowser.DKStartIndex = 2
        
        self.presentViewController(modalImageBrowser, animated: true, completion: nil)
    }

```

## Customization
- `DKBackgroundColor` Background color of the image browser. The default is white.
- `DKImagePadding` Padding around the images. The default is 20.
- `DKImageWidth` Width of the main image being displayed. The default is the width of the screen -100.
- `DKStartIndex` Starting image index. The default is 0 (first image).
- `DKThumbnailStripHeight` Height of the thumbnail strip. The default is 100.
- `DKThumbnailStripPosition`  Position of the thumbnail strip, either `DKThumbnailStripPositionTop` or `DKThumbnailStripPositionBottom`. The default is `DKThumbnailStripPositionBottom`.

# Demo
DKImageBrowser includes Swift and Objective-C demo projects.

# Compatibility
- This project uses ARC.
- This project supports iPhone and iPad.
- This project was tested with iOS 8.

# Contact
- [github.com/dkhamsing](https://github.com/dkhamsing)
- [twitter.com/dkhamsing](https://twitter.com/dkhamsing)
- [Send a message](http://dkhamsing.tumblr.com/ask)

# License
DKImageBrowser is available under the MIT license. See the [LICENSE](LICENSE) file for more info.


CHANGELOG.md
# DKImageBrowser Changelog

https://github.com/dkhamsing/DKImageBrowser/releases

## 0.1.5

- Add changelog
- Use `NSURLSession` 
- Xcode 7
- Swift example

## 0.1.4

- Simplify loading images from URL

## 0.1.3

- Add start index property
- Update logging

## 0.1.2

- CocoaPods


## 0.1.1

- Update documentation
- Tests

## 0.1

- Initial version



Demo/Demo/DKViewController.m
//
//  DKViewController.m
//  Demo
//
//  Created by dkhamsing on 3/20/14.
//
//

#import "DKViewController.h"
#import "DKImageBrowser.h"
#import "DKModalImageBrowser.h"
 

@implementation DKViewController

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil {
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        self.title = @"Demo";
        
        self.navigationItem.leftBarButtonItem = [[UIBarButtonItem alloc] initWithTitle:@"Default" style:UIBarButtonItemStylePlain target:self action:@selector(actionSimpleDefault)];
        
        self.navigationItem.rightBarButtonItem = [[UIBarButtonItem alloc] initWithTitle:@"Custom" style:UIBarButtonItemStylePlain target:self action:@selector(actionCustom)];
        
        UILabel *creditsLabel = [[UILabel alloc] initWithFrame:CGRectMake(0, 80, self.view.bounds.size.width, 80)];
        creditsLabel.textAlignment = NSTextAlignmentCenter;
        creditsLabel.text = @"DKImageBrowser";
        [self.view addSubview:creditsLabel];
    }
    return self;
}


- (void)actionSimpleDefault {
    // set up data source with UImage objects
    NSArray *imageDataSource = @[ [UIImage imageNamed:@"400-1.jpg"],
                                  [UIImage imageNamed:@"400-2.jpg"],
                                  ];
    
    // default
    DKImageBrowser *imageBrowser = [[DKImageBrowser alloc] init];
    imageBrowser.title = @"😺 UIImage Data Source";
    imageBrowser.DKImageDataSource = imageDataSource;
    
    // push on navigation stack
    [self.navigationController pushViewController:imageBrowser animated:YES];
}


- (void)actionCustom {
    // set up data source with image URLs, thank you http://placekitten.com
    NSMutableArray *kittenDataSource = [[NSMutableArray alloc]init];
    NSInteger imageHeight = 416;
    NSNumber *imageWidth = @(self.view.bounds.size.width - 20);
    for (int i=imageWidth.integerValue; i<(imageWidth.integerValue+15); i++) {
        NSString *placeKitten =
        [NSString stringWithFormat: @"http://placekitten.com/%.0f/%d", i*1.1, imageHeight];
        [kittenDataSource addObject:placeKitten];
    }
    
    // present modally
    DKModalImageBrowser *modalImageBrowser = [[DKModalImageBrowser alloc] init];
    modalImageBrowser.title = @"😺 URL Data Source";
 
    // customize
    modalImageBrowser.imageBrowser.DKImageDataSource = [kittenDataSource copy];
    modalImageBrowser.imageBrowser.DKImagePadding = 5;
    modalImageBrowser.imageBrowser.DKImageWidth = imageWidth.floatValue;
    modalImageBrowser.imageBrowser.DKThumbnailStripHeight = 110;
    modalImageBrowser.imageBrowser.DKBackgroundColor = [UIColor blackColor];
    modalImageBrowser.imageBrowser.DKThumbnailStripPosition = DKThumbnailStripPositionTop;
    
    [self presentViewController:modalImageBrowser animated:YES completion:nil];    
}


@end
